VideoLocker/src/main/java/org/edx/mobile/view/CourseDiscussionPostsThreadFragment.java

```diff
@@ -197,15 +216,15 @@ public void onEventMainThread(DiscussionCommentPostedEvent event) {
 
     @SuppressWarnings("unused")
     public void onEventMainThread(DiscussionThreadPostedEvent event) {
-        // If a new post is created in this topic, insert it at the top of the list, after any pinned posts
-        if (discussionTopic.containsThread(event.getDiscussionThread())) {
-            int i = 0;
-            for (; i < discussionPostsAdapter.getCount(); ++i) {
-                if (!discussionPostsAdapter.getItem(i).isPinned()) {
-                    break;
-                }
+        DiscussionThread newThread = event.getDiscussionThread();
+        // If a new post is created in this topic, insert it at the top of the list
+        if (discussionTopic.containsThread(newThread)) {
+            if (postsFilter == DiscussionPostsFilter.UNANSWERED &&
+                    newThread.getType() != DiscussionThread.ThreadType.QUESTION) {
+                return;
             }
-            discussionPostsAdapter.insert(event.getDiscussionThread(), i);
+
+            discussionPostsAdapter.insert(newThread, 0);
```

Comment:
Hmm, actually the previous implementation seems fine here. We shouldn't be adding new threads on top of pinned threads.